### DISCUSSION TITLE
Queue Batch / Collapse / Combine

### DISCUSSION THREAD

**Discussion by dillingham** [laravel/framework#32945] [General]
Some APIs have an option for batch handling.. like storing 20 records.
This, along with cache proxies, are to avoid hitting rate limits. 
A queue mechanism for collapsing jobs into one related job could be cool.
Like 5 `ApiCreateJobs` within 5 seconds turns into `BatchApiCreateJob`

**Comment by dillingham** [comment]
Pretty heavily discussed here actually

https://github.com/laravel/ideas/issues/410

**Comment by nyrik8989** [comment]
Who tried to implement a similar solution in Laravel ???

https://jolicode.com/blog/batching-symfony-messenger-messages

```
pcntl_async_signals(true);
pcntl_signal(SIGALRM, \Closure::fromCallable([$this, 'batchBuffer']));
